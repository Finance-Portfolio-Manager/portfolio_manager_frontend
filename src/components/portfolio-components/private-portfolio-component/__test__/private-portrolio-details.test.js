import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PrivatePortfolioDetails from './../private-portfolio-details';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

let stock1 = {stockName:"Amazon", stockQuantity:10, stockAveragePrice:3300, stockCurrentPrice:3400, stockChange:8};
let stock2 = {stockName:"Tesla", stockQuantity:15, stockAveragePrice:700, stockCurrentPrice:715, stockChange:2};
let stock3 = {stockName:"AMC", stockQuantity:20, stockAveragePrice:10, stockCurrentPrice:33, stockChange:230};
let stock4 = {stockName:"GME", stockQuantity:5, stockAveragePrice:40, stockCurrentPrice:180, stockChange:230};
let stock5 = {stockName:"LYFT", stockQuantity:7, stockAveragePrice:30, stockCurrentPrice:120, stockChange:230};
let stock6 = {stockName:"UBER", stockQuantity:15, stockAveragePrice:50, stockCurrentPrice:200, stockChange:230};
let stockList = [stock1, stock2, stock3, stock4, stock5, stock6];

let stockValues = stockList.map((stock) => stock.stockQuantity * stock.stockCurrentPrice);
let portfolioValue = stockValues.reduce((sum, currentStockValue) =>  sum + currentStockValue );

let portObj1 = {id: 1, name:"Portfolio 1", value:portfolioValue, stocks: stockList};
let portObj2 = {id: 2, name:"Portfolio 2", value:portfolioValue, stockList: stockList};
let portObj3 = {id: 3, name:"Portfolio 3", value:portfolioValue, stockList: stockList};
let portObj4 = {id: 4, name:"Portfolio 4", value:portfolioValue, stockList: stockList};
let portObj5 = {id: 5, name:"Portfolio 5", value:portfolioValue, stockList: stockList};
let portObj6 = {id: 6, name:"Portfolio 6", value:portfolioValue, stockList: stockList};

let portArray = [portObj1, portObj2, portObj3, portObj4, portObj5, portObj6];

const toggleFn = jest.fn(); 

configure({ adapter: new Adapter() });
const wrapper = shallow(<PrivatePortfolioDetails portfolio={portObj1} username="test_user" toggleShowDetails={toggleFn}></PrivatePortfolioDetails>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<PrivatePortfolioDetails portfolio={portObj1} username="test_user" toggleShowDetails={toggleFn}></PrivatePortfolioDetails>).toJSON();
    expect(tree).toMatchSnapshot();
})