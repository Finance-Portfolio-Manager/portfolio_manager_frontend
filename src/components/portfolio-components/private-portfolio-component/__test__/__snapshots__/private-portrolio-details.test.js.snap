// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <h3>
      Portfolio 1
    </h3>
  </div>,
  <div
    className="row"
  >
    <div
      className="folioContent col-md-3"
    >
      <div>
        <canvas
          data-testid="canvas"
          height={150}
          onClick={[Function]}
          role="img"
          width={300}
        />
      </div>
    </div>
    <div
      className="folioContent col"
    >
      <div
        className="wrapper fadeInDown folioContent mx-3"
        id="portfolio"
      >
        <div
          id="portfolio-values"
        >
          <h4
            id="portfolio-total"
          />
          <div
            id="portfolio-change"
          />
          <div
            id="dollar-change"
          />
        </div>
        <div
          className="table-responsive secondary-text"
        >
          <div
            className="table-responsive"
          >
            <table
              className="table borderless table-hover secondary-text table"
              id="stocks-view"
            >
              <thead>
                <tr>
                  <th
                    scope="col"
                  >
                    Stock
                  </th>
                  <th
                    scope="col"
                  >
                    Quantity
                  </th>
                  <th
                    scope="col"
                  >
                    Average Buy Price
                  </th>
                  <th
                    scope="col"
                  >
                    Current Price
                  </th>
                  <th
                    scope="col"
                  >
                    P/L
                  </th>
                </tr>
              </thead>
              <tbody
                id="stocks-body"
              >
                <tr>
                  <td />
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
                <tr>
                  <td />
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
                <tr>
                  <td />
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
                <tr>
                  <td />
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
                <tr>
                  <td />
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
                <tr>
                  <td />
                  <td />
                  <td />
                  <td />
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
          <button
            className="btn btn-primary"
            disabled={false}
            type="button"
          >
            Hide Details
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
